src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java

```diff
@@ -72,77 +76,103 @@ public class WebSocketClient<T> extends WebSocketListener {
 
     protected void connect(Set<String> newChannels) {
         if (this.state != WSState.CONNECTED && this.state != WSState.CONNECTING) {
-            log.debug("{} Connecting...", logPrefix);
+            log.debug("{} Connecting to channels {} ...", logPrefix, newChannels);
             this.state = WSState.CONNECTING;
             this.webSocket = apiClient.createWebSocket(this.connectionRequest, this);
+            if (this.webSocket == null) {
+                log.error("Error connect {}", this.connectionRequest);
+            }
         } else {
-            log.warn("{} Already connected", logPrefix);
+            log.warn("{} Already connected to channels {}", logPrefix, channels);
         }
         if (!newChannels.isEmpty()) {
             this.subscribe(newChannels);
-            this.scheduledPingTask = this.scheduler.scheduleAtFixedRate(new PingTask(newChannels),
-                    WEB_SOCKET_KEEP_ALIVE_TOPIC_PERIOD_SEC, WEB_SOCKET_KEEP_ALIVE_TOPIC_PERIOD_SEC, TimeUnit.SECONDS);
+            //create ping task for keep alive WS connection if no messages were produced be server
+            this.scheduledPingTask = scheduler.scheduleAtFixedRate(new PingTask(),
+                    this.config.getWebSocketChannelKeepAlivePeriodSec(), this.config.getWebSocketChannelKeepAlivePeriodSec(), TimeUnit.SECONDS);
         }
+        currentReconnectAttemptIndex.set(0);//reset reconnection indexer due to successful connection
     }
 
-    public void send(String str) {
+    public void sendRequest(String request) {
         boolean result = false;
-        log.debug("{}:send() {}", logPrefix, str);
+        log.debug("{} Try to send request {}", logPrefix, request);
```

Comment:
what about ping requests? they're also going to be logged as DEBUG. In contrast, in line 124 we log it as TRACE. In the current state of code, ping requests going to be logged twice: as DEBUG as well as TRACE

Reply:
Yeah, if you set logging level 'TRACE' for our library you will get log message for ping request twice. It was done only for detailed debugging "keep alive" functionality. If it's so matter, we can delete additional logging in ping() method.